### ISSUE TITLE
SQL: NPE while using HISTOGRAM with sum of intervals

### ISSUE THREAD

**Issue by astefan** [elastic/elasticsearch#38306]
```
sql> select histogram("@timestamp", INTERVAL 1 MINUTE + INTERVAL 2 MINUTE) as h from test group by h;                                                                     
Server error [Server encountered an error [null]. [java.lang.NullPointerException                                                                                         
        at org.elasticsearch.xpack.sql.planner.QueryFolder$FoldAggregate.rule(QueryFolder.java:350)                                                                       
        at org.elasticsearch.xpack.sql.planner.QueryFolder$FoldAggregate.rule(QueryFolder.java:205)                                                                       
        at org.elasticsearch.xpack.sql.tree.Node.lambda$transformUp$11(Node.java:196)                                                                                     
        at org.elasticsearch.xpack.sql.tree.Node.transformUp(Node.java:190)                                                                                               
        at org.elasticsearch.xpack.sql.tree.Node.transformUp(Node.java:196)                                                                                               
        at org.elasticsearch.xpack.sql.planner.QueryFolder$FoldingRule.apply(QueryFolder.java:582)                                                                        
        at org.elasticsearch.xpack.sql.planner.QueryFolder$FoldingRule.apply(QueryFolder.java:578)                                                                        
        at org.elasticsearch.xpack.sql.rule.RuleExecutor$Transformation.<init>(RuleExecutor.java:82)                                                                      
        at org.elasticsearch.xpack.sql.rule.RuleExecutor.executeWithInfo(RuleExecutor.java:158)                                                                           
        at org.elasticsearch.xpack.sql.rule.RuleExecutor.execute(RuleExecutor.java:130)                                                                                   
        at org.elasticsearch.xpack.sql.planner.QueryFolder.fold(QueryFolder.java:86)                                                                                      
        at org.elasticsearch.xpack.sql.planner.Planner.foldPlan(Planner.java:38)                                                                                          
        at org.elasticsearch.xpack.sql.planner.Planner.plan(Planner.java:28)                                                                                              
        at org.elasticsearch.xpack.sql.session.SqlSession.lambda$physicalPlan$4(SqlSession.java:156)                                                                      
        at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:61)                                                                                   
        at org.elasticsearch.xpack.sql.session.SqlSession.lambda$optimizedPlan$3(SqlSession.java:152)                                                                     
        at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:61)                                                                                   
        at org.elasticsearch.xpack.sql.session.SqlSession.lambda$preAnalyze$2(SqlSession.java:140)                                                                        
        at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:61)                                                                                   
        at org.elasticsearch.xpack.sql.analysis.index.IndexResolver.lambda$resolveAsMergedMapping$3(IndexResolver.java:245)                                               
        at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:61)                                                                                   
        at org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:68)                                                                         
        at org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:64)                                                                         
        at org.elasticsearch.action.fieldcaps.TransportFieldCapabilitiesAction.lambda$doExecute$1(TransportFieldCapabilitiesAction.java:84)                               
        at org.elasticsearch.action.fieldcaps.TransportFieldCapabilitiesAction$1.onResponse(TransportFieldCapabilitiesAction.java:97)                                     
        at org.elasticsearch.action.fieldcaps.TransportFieldCapabilitiesAction$1.onResponse(TransportFieldCapabilitiesAction.java:93)                                     
        at org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:68)                                                                         
        at org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:64)                                                                         
        at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleResponse(TransportSingleShardAction.java:271)               
        at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2.handleResponse(TransportSingleShardAction.java:255)               
        at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleResponse(TransportService.java:1108)                                          
        at org.elasticsearch.transport.TransportService$DirectResponseChannel.processResponse(TransportService.java:1189)                                                 
        at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1169)                                                    
        at org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:54)                                                                    
        at org.elasticsearch.action.support.ChannelActionListener.onResponse(ChannelActionListener.java:47)                                                               
        at org.elasticsearch.action.support.ChannelActionListener.onResponse(ChannelActionListener.java:30)                                                               
        at org.elasticsearch.action.support.single.shard.TransportSingleShardAction$1.doRun(TransportSingleShardAction.java:117)                                          
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:751)                                         
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)                                                                        
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)                                                                      
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)                                                                      
        at java.base/java.lang.Thread.run(Thread.java:834)                                                                                                                
```

**Comment by costin** [comment]
The issue here is that the functions are identical however new ids get created when folding the argument so now the projection and the grouping don't match anymore.
This could potentially be solved by removing duplicated function or by keeping the grouping reference in place instead of copying it over.
However I think it might be solved better with the attribute narrowing.

**Comment by matriv** [comment]
This has been fixed by #49570

**Comment by matriv** [comment]
This has been fixed by #49570
